Sources/DatabaseKit/ConnectionPool/DatabaseConnectionPool.swift

```diff
@@ -58,7 +59,13 @@ public final class DatabaseConnectionPool<Database> where Database: DatabaseKit.
     public func withConnection<T>(_ closure: @escaping (Database.Connection) throws -> Future<T>) -> Future<T> {
         let release = releaseConnection
         return requestConnection().flatMap(to: T.self) { conn in
-            return try closure(conn).always { release(conn) }
+            do {
+                return try closure(conn).always { release(conn) }
+            } catch {
+                release(conn)
+
+                throw error
+            }
```

Comment:
this fix is also required in `Container+NewConnection.swift`, here https://github.com/vapor/database-kit/blob/master/Sources/DatabaseKit/Connection/Container%2BNewConnection.swift#L20

Without it, the connection is never release if `closure` throws.